plugins/html-plugin/src/main/java/ru/sbtqa/tag/pagefactory/find/HtmlFindUtils.java

```diff
@@ -0,0 +1,382 @@
+
+            if (!(wait || element.isPresent())) {
+                throw new IllegalStateException(errorText);
+            } else {
+                throw new ElementSearchException(errorText);
+            }
+        }
+        if (wait) {
+            Wait.visibility(HtmlElementUtils.getWebElement(element.getElement()),
+                    "The page does not display the element " + formErrorMessage(element));
+        }
+        return (T) element.getElement();
+    }
+
+    /**
+     * Gets a list of elements by name or path
+     * <p>
+     * As a path, an enumeration of elements in a nested structure can be used
+     * through the separator "{@code ->}", for example:
+     * <ul>
+     * <li> element in the block: Block{@code ->}Element
+     * <li> element in the list with the sequence number 3: List{@code ->}3
+     * </ul>
+     * <p>
+     *
+     * @param <T> type of the returned element
+     * @param name list name or path to it
+     * @return Returns a list from a page by name or path
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.ElementDescriptionException
+     */
+    public <T extends WebElement> List<T> findList(String name) throws ElementDescriptionException {
+        return findList(null, name);
+    }
+    
+    /**
+     * Gets a list of elements by name or path
+     * <p>
+     * As a path, an enumeration of elements in a nested structure can be used
+     * through the separator "{@code ->}", for example:
+     * <ul>
+     * <li> element in the block: Block{@code ->}Element
+     * <li> element in the list with the sequence number 3: List{@code ->}3
+     * </ul>
+     * <p>
+     *
+     * @param <T> type of the returned element
+     * @param context current context - where to start searching for an element.
+     * Specify {@code null} if you need to perform a search in the context of
+     * the current page, or an element to search for
+     * @param name list name or path to it
+     * @return Returns a list from a page by name or path
+     * @throws ru.sbtqa.tag.pagefactory.exceptions.ElementDescriptionException
+     */
+    public <T extends WebElement> List<T> findList(T context, String name) throws ElementDescriptionException {
+        ComplexElement element = new ComplexElement(context, name, false);
+        Field field;
+
+        if (name.contains(ELEMENT_SEPARATOR)) {
+            int lastSeparatorIndex = name.lastIndexOf(ELEMENT_SEPARATOR);
+
+            element = findComplexElement(context, name.substring(0, lastSeparatorIndex), false);
+            String listName = name.substring(lastSeparatorIndex + ELEMENT_SEPARATOR.length(), name.length());
+
+            field = getField(element.getElement(), listName);
+            if (field == null) {
+                throw new ElementSearchException("Element list not found");
+            }
+        } else {
+            field = getField(context, name);
+        }
+
+        if (!field.getType().isAssignableFrom(List.class)) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: A "NullPointerException" could be thrown; "field" is nullable here. ([squid:S2259](https://sonarcloud.io/organizations/sbtqasonarkey/rules#rule_key=squid%3AS2259))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=ru.sbtqa.tag%3Apage-factory-2&pullRequest=133&issues=AWfPt2o64tUZZzFFj6HW&open=AWfPt2o64tUZZzFFj6HW)

[ISSUE_KEY]: AWfPt2o64tUZZzFFj6HW